editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/ArgumentInputDialog.java

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.camel.editor.globalconfiguration.beans;
+
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.FocusEvent;
+import org.eclipse.swt.events.FocusListener;
+import org.eclipse.swt.events.KeyEvent;
+import org.eclipse.swt.events.KeyListener;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+
+/**
+ * @author bfitzpat
+ * 
+ */
+public class ArgumentInputDialog extends TitleAreaDialog {
+
+    private Text argumentTypeText;
+    private Text argumentValueText;
+
+    private String argumentType = null;
+    private String argumentValue = null;
+
+    /**
+     * constructor.
+     * 
+     * @param parent the parent
+     */
+    public ArgumentInputDialog(Shell parent) {
+        super(parent);
+    }
+
+    @Override
+    protected Control createDialogArea(Composite parent) {
+        setTitle("Argument Details");
+        setMessage("Specify type and value details for the new argument.");
+        getShell().setText("Bean Argument");
+
+        Composite area = new Composite(parent, SWT.NULL);
+        GridLayout gridLayout = new GridLayout(2, false);
+        area.setLayout(gridLayout);
+        area.setLayoutData(new GridData(GridData.FILL_BOTH));
+        argumentTypeText = createLabelAndText(area, "Type");
```

Comment:
I will go through an internationalization pass as soon as we figure out some of the pending questions for this first pass at adding the functionality.